aboutcitus/concepts.rst

```diff
@@ -0,0 +1,125 @@
+
+Suppose that shard 102027 is associated with the row in question. This means the row should be read or written to a table called ``github_events_102027`` in one of the workers. Which worker? That is determined entirely by the metadata tables, and the mapping of shard to worker is known as the shard *placement*.
+
+Joining some :ref:`metadata tables <metadata_tables>` gives us the answer. These are the types of lookups that the coordinator does to route queries. It rewrites queries into fragments that refer to the specific tables like ``github_events_102027``, and runs those fragments on the appropriate workers.
+
+.. code-block:: sql
+
+  SELECT
+      shardid,
+      node.nodename,
+      node.nodeport
+  FROM pg_dist_placement placement
+  JOIN pg_dist_node node
+    ON placement.groupid = node.groupid
+   AND node.noderole = 'primary'::noderole
+  WHERE shardid = 102027;
+
+::
+
+  ┌─────────┬───────────┬──────────┐
+  │ shardid │ nodename  │ nodeport │
+  ├─────────┼───────────┼──────────┤
+  │  102027 │ localhost │     5433 │
+  └─────────┴───────────┴──────────┘
+
+In our example of ``github_events`` there were four shards. The number of shards is configurable per table at the time of its distribution across the cluster. The best choice of shard count depends on your use case, see :ref:`prod_shard_count`.
+
+Finally note that Citus allows shards to be replicated for protection against data loss. There are two replication "modes:" Citus replication and streaming replication. The former creates extra backup shard placements and runs queries against all of them that update any of them. The latter is more efficient and utilizes PostgreSQL's streaming replication to back up the entire database of each node to a follower database. This is transparent and does not require the involvement of Citus metadata tables.
+
+Co-Location
+-----------
+
+Since shards and their replicas can be placed on nodes as desired, it makes sense to place shards containing related rows of related tables together on the same nodes. That way join queries between them can avoid sending as much information over the network, and can be performed inside a single Citus node.
+
+One example is a database with stores, products, and purchases. If all three tables contain -- and are distributed by -- a store_id column, then all queries restricted to a single store can run efficiently on a single worker node. This is true even when the queries involve any combination of these tables.
+
+For a full explanation and examples of this concept, see :ref:`colocation`.
+
+Parallelism
+-----------
+
+Spreading queries across multiple machines allows more queries to run at once, and allows processing speed to scale by adding new machines to the cluster. Additionally splitting a single query into fragments as described in the previous section boosts the processing power devoted to it. The latter situation achieves the greatest *parallelism,* meaning utilization of CPU cores.
```

Comment:
Let's link to our Query Processing section. It's a bit out of date, but we will update that soon.